src/gpio.rs

```diff
@@ -127,6 +130,127 @@ pub struct Disabled;
 
 pub struct Unknown;
 
+#[cfg(all(not(feature = "riscv-ulp-hal"), feature = "std"))]
+pub struct Subscribed;
```

Comment:
We really need to triple-check that a pin which is in input-output mode really cannot be subscribed to. Can you look in the ESP-IDF documentation? Otherwise we might have the wrong implementation here.

Also, how about renaming `Subscribed` to `SubscribedInput` as it is a tad clearer here that the pin is still configured as an input pin (right?). This is really important, as folks will surely want to use regular `is_low` / `is_high` on input pins which are in subscribed state.

Reply:
1. I already checked the documentation ( [here](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/gpio.html?highlight=gpio_intr#_CPPv416gpio_intr_enable10gpio_num_t), [here](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/gpio.html?highlight=gpio_intr#_CPPv411gpio_mode_t) and [here](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/gpio.html?highlight=gpio_intr#_CPPv418gpio_set_direction10gpio_num_t11gpio_mode_t) for example), but I was not able to find any info on this, that's why I resorted to test it. I will have a second look in more general doc though.

2. You're right `SubscribedInput` is way more clear and I missed the `hal` InputPin impl